packages/fs-detectors/test/unit.framework-detector.test.ts

```diff
@@ -135,11 +135,18 @@ describe('DetectorFilesystem', () => {
     };
 
     const fs = new VirtualFilesystem(files);
+    const hasPathSpy = jest.spyOn(fs, '_hasPath');
 
-    expect(await fs.readdir('/')).toEqual([
+    expect(await fs.readdir('/', { potentialFiles: ['config.rb'] })).toEqual([
```

Comment:
What about adding `potentialFiles` with `packages/app1/package.json`? I think we need more test coverage here